bundles/org.openhab.binding.monopriceaudio/src/main/java/org/openhab/binding/monopriceaudio/internal/handler/MonopriceAudioHandler.java

```diff
@@ -0,0 +1,723 @@
+import org.openhab.binding.monopriceaudio.internal.communication.MonopriceAudioCommand;
+import org.openhab.binding.monopriceaudio.internal.communication.MonopriceAudioConnector;
+import org.openhab.binding.monopriceaudio.internal.communication.MonopriceAudioDefaultConnector;
+import org.openhab.binding.monopriceaudio.internal.communication.MonopriceAudioIpConnector;
+import org.openhab.binding.monopriceaudio.internal.communication.MonopriceAudioMessageEvent;
+import org.openhab.binding.monopriceaudio.internal.communication.MonopriceAudioMessageEventListener;
+import org.openhab.binding.monopriceaudio.internal.communication.MonopriceAudioSerialConnector;
+import org.openhab.binding.monopriceaudio.internal.communication.MonopriceAudioZone;
+import org.openhab.binding.monopriceaudio.internal.configuration.MonopriceAudioThingConfiguration;
+import org.openhab.binding.monopriceaudio.internal.dto.MonopriceAudioZoneDTO;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link MonopriceAudioHandler} is responsible for handling commands, which are sent to one of the channels.
+ *
+ * Based on the Rotel binding by Laurent Garnier
+ *
+ * @author Michael Lobstein - Initial contribution
+ */
+@NonNullByDefault
+public class MonopriceAudioHandler extends BaseThingHandler implements MonopriceAudioMessageEventListener {
+    private static final long RECON_POLLING_INTERVAL = TimeUnit.SECONDS.toSeconds(60);
+    private static final long INITIAL_POLLING_DELAY = TimeUnit.SECONDS.toSeconds(5);
+    private static final Pattern PATTERN = Pattern
+            .compile("^(\\d{2})(\\d{2})(\\d{2})(\\d{2})(\\d{2})(\\d{2})(\\d{2})(\\d{2})(\\d{2})(\\d{2})(\\d{2})");
+
+    private static final String ZONE = "ZONE";
+    private static final String ALL = "all";
+    private static final String CHANNEL_DELIMIT = "#";
+    private static final String ON = "01";
+    private static final String OFF = "00";
+
+    private static final int ONE = 1;
+    private static final int MAX_ZONES = 18;
+    private static final int MAX_SRC = 6;
+    private static final int MIN_VOLUME = 0;
+    private static final int MAX_VOLUME = 38;
+    private static final int MIN_TONE = -7;
+    private static final int MAX_TONE = 7;
+    private static final int MIN_BALANCE = -10;
+    private static final int MAX_BALANCE = 10;
+    private static final int BALANCE_OFFSET = 10;
+    private static final int TONE_OFFSET = 7;
+
+    private final Logger logger = LoggerFactory.getLogger(MonopriceAudioHandler.class);
+    private final MonopriceAudioStateDescriptionOptionProvider stateDescriptionProvider;
+
+    private @Nullable ScheduledFuture<?> reconnectJob;
+    private @Nullable ScheduledFuture<?> pollingJob;
+
+    private SerialPortManager serialPortManager;
+    private MonopriceAudioConnector connector = new MonopriceAudioDefaultConnector();
+
+    private ArrayList<String> ignoreZones = new ArrayList<>();
```

Comment:
Since order doesn't matter and you are only performing lookups on it, use a Set instead.
```suggestion
    private final Set<String> ignoreZones = new HasSet<>();
```